loadbalancer/zk/client.go

```diff
@@ -0,0 +1,157 @@
+package zk
+
+import (
+	"errors"
+	"net"
+	"strings"
+	"time"
+
+	"github.com/eapache/channels"
+	"github.com/samuel/go-zookeeper/zk"
+
+	"github.com/go-kit/kit/log"
+)
+
+// DefaultACL is the default ACL to use for creating znodes
+var DefaultACL = zk.WorldACL(zk.PermAll)
+
+const (
+	// ConnectTimeout is the default timeout to establish a connection to a
+	// ZooKeeper node.
+	ConnectTimeout = 2 * time.Second
+	// SessionTimeout is the default timeout to keep the current ZooKeeper
+	// session alive during a temporary disconnect.
+	SessionTimeout = 5 * time.Second
+)
+
+// Client is a wrapper around a lower level ZooKeeper client implementation.
+type Client interface {
+	// GetEntries should query the provided path in ZooKeeper, place a watch on
+	// it and retrieve data from its current child nodes.
+	GetEntries(path string) ([]string, channels.SimpleOutChannel, error)
```

Comment:
Yes that was my exact idea so we have a loose coupling and ability to have an interface that could be implemented using other underlying libraries. If we don't want that we can skip having an interface all together as there is only one client implementation compatible with the publisher.